include/seqan3/argument_parser/argument_parser.hpp

```diff
@@ -425,6 +425,62 @@ class argument_parser
         return *sub_parser;
     }
 
+    /*!\brief Checks whether the **short** option identifier (`id`) was set on the command line by the user.
+     * \param[in] id The short option identifier to search for.
+     * \returns `true` if option identifier `id` was set on the command line by the user.
+     *
+     * \details
+     *
+     * You can only ask for option identifiers that were added to the parser beforehand via
+     * seqan3::argument_parser::add_option.
+     *
+     * ### Example:
+     *
+     * \include test/snippet/argument_parser/is_option_set.cpp
+     */
+    bool is_option_set(char const id) const
+    {
+        if (!parse_was_called)
+            throw design_error("You can only asked which options have been set after calling the function `parse()`.");
+
+        if (std::find(used_option_ids.begin(), used_option_ids.end(), std::string{id}) == used_option_ids.end())
+            throw design_error("You can only ask for option identifiers that you added with add_option() before.");
+
+        // we only need to search for an option before the `end_of_options_indentifier` (`--`)
+        auto end_of_options = std::find(cmd_arguments.begin(), cmd_arguments.end(), end_of_options_indentifier);
+        return detail::format_parse::find_option_id(cmd_arguments.begin(), end_of_options, id) != end_of_options;
+    }
+
+    /*!\brief Checks whether the **long** option identifier (`id`) was set on the command line by the user.
+     * \param[in] id The long option identifier to search for (must not contain dashes!).
+     * \returns `true` if option identifier `id` was set on the command line by the user.
+     *
+     * \details
+     *
+     * You can only ask for option identifiers that were added to the parser beforehand via
+     * seqan3::argument_parser::add_option.
+     *     *
+     * ### Example:
+     *
+     * \include test/snippet/argument_parser/is_option_set.cpp
+     */
+    bool is_option_set(std::string const & id) const
+    {
+        if (!parse_was_called)
+            throw design_error{"You can only asked which options have been set after calling the function `parse()`."};
+
+        if (id.size() == 1)
+            throw design_error{"Long option identifiers must be longer than one character! If " + id + "' was meant to "
+                               "be a short identifier, please pass it as a char ('') not a string (\"\")!"};
+
+        if (std::find(used_option_ids.begin(), used_option_ids.end(), id) == used_option_ids.end())
+            throw design_error{"You can only ask for option identifiers that you added with add_option() before."};
+
+        // we only need to search for an option before the `end_of_options_indentifier` (`--`)
+        auto end_of_options = std::find(cmd_arguments.begin(), cmd_arguments.end(), end_of_options_indentifier);
+        return detail::format_parse::find_option_id(cmd_arguments.begin(), end_of_options, id) != end_of_options;
+    }
```

Comment:
```suggestion
    bool is_option_set(char const id) const
    {
        return is_option_set_impl(std::string{id}, false);
    }
   
    bool is_option_set(std::string const & id) const
    {
         return is_option_set_impl(id, true);
    }

    bool is_option_set_impl(std::string const & id, bool const is_long_option_id) const
    {
        if (!parse_was_called)
            throw design_error{"You can only ask which options have been set after calling the function `parse()`."};

        if (is_long_option_id && id.size() == 1)
            throw design_error{"Long option identifiers must be longer than one character! If " + id + "' was meant to "
                               "be a short identifier, please pass it as a char ('') not a string (\"\")!"};

        if (std::find(used_option_ids.begin(), used_option_ids.end(), id) == used_option_ids.end())
            throw design_error{"You can only ask for option identifiers that you added with add_option() before."};

        // we only need to search for an option before the `end_of_options_indentifier` (`--`)
        auto end_of_options = std::find(cmd_arguments.begin(), cmd_arguments.end(), end_of_options_indentifier);
        return detail::format_parse::find_option_id(cmd_arguments.begin(), end_of_options, id) != end_of_options;
    }
```

Reply:
This also also reduces the typo to only one occurrence :)

Reply:
This does not work because 
```
is_option_set_impl(std::string{id}, false);
```
will trigger the exception of long ids, that they must be longer than one char.

Reply:
Wait what? I did care of this by passing this additional bool parameter to the internal interface. See, the check whether the long_id has only one element is only evaluate if `is_long_option_id` is set to true. But it is not in the case of a short id. Other than that the code was exactly the same.

Reply:
Oh, no. This goes on in the parse operations as well. I see.